core/src/components/label/label.scss

```diff
@@ -19,18 +19,31 @@
 
   text-overflow: ellipsis;
 
-  white-space: nowrap;
-
+  // Overflow hidden is required for the proper
+  // margins between skeleton text elements
   overflow: hidden;
   box-sizing: border-box;
 }
 
+// TODO(FW-5289): remove
+:host-context(.item-legacy) {
+  white-space: nowrap;
+}
+
+// TODO(FW-5289): move to :host-context(.item)
+// Shouldn't need :not(.item-input) as this was
+// only needed because of the specificity with
+// :not(.item-legacy)
+:host-context(.item:not(.item-input):not(.item-legacy)) {
```

Comment:
How is this going to work in Safari/Firefox? They don't support `:host-context` IIRC

Reply:
See my comment here: https://github.com/ionic-team/ionic-framework/pull/28146#discussion_r1357307512